clang/lib/Sema/SemaExpr.cpp

```diff
@@ -20760,7 +20761,8 @@ ExprResult RebuildUnknownAnyExpr::VisitCallExpr(CallExpr *E) {
 
 ExprResult RebuildUnknownAnyExpr::VisitObjCMessageExpr(ObjCMessageExpr *E) {
   // Verify that this is a legal result type of a call.
-  if (DestType->isArrayType() || DestType->isFunctionType()) {
+  if ((DestType->isArrayType() && !S.Context.isReturnableArrayType()) ||
```

Comment:
This code shouldn't be updated. We're definitely not building ObjCMessage expressions in HLSL!